zio-schema-thrift/shared/src/main/scala/zio/schema/codec/ThriftCodec.scala

```diff
@@ -0,0 +1,1090 @@
+          p.writeByte(v.getValue.toByte)
+        case (StandardType.MonthDay, v: MonthDay) =>
+          encodeRecord(None, monthDayStructure(), ListMap("month" -> v.getMonthValue, "day" -> v.getDayOfMonth))
+        case (StandardType.Period, v: Period) =>
+          encodeRecord(
+            None,
+            periodStructure(),
+            ListMap("years" -> v.getYears, "months" -> v.getMonths, "days" -> v.getDays)
+          )
+        case (StandardType.Year, v: Year) =>
+          p.writeI32(v.getValue)
+        case (StandardType.YearMonth, v: YearMonth) =>
+          encodeRecord(None, yearMonthStructure(), ListMap("year" -> v.getYear, "month" -> v.getMonthValue))
+        //        case (StandardType.ZoneId, v: ZoneId) =>
+        //          encodePrimitive(fieldNumber, StandardType.StringType, v.getId)
+        //        case (StandardType.ZoneOffset, v: ZoneOffset) =>
+        //          encodePrimitive(fieldNumber, StandardType.IntType, v.getTotalSeconds)
+        //        case (StandardType.Duration(_), v: Duration) =>
+        //          encodeRecord(fieldNumber, durationStructure(), ListMap("seconds" -> v.getSeconds, "nanos" -> v.getNano))
+        case (StandardType.Instant(formatter), v: Instant) =>
+          p.writeString(formatter.format(v))
+        case (StandardType.LocalDate(formatter), v: LocalDate) =>
+          p.writeString(formatter.format(v))
+        case (StandardType.LocalTime(formatter), v: LocalTime) =>
+          p.writeString(formatter.format(v))
+        case (StandardType.LocalDateTime(formatter), v: LocalDateTime) =>
+          p.writeString(formatter.format(v))
+        case (StandardType.OffsetTime(formatter), v: OffsetTime) =>
+          p.writeString(formatter.format(v))
+        case (StandardType.OffsetDateTime(formatter), v: OffsetDateTime) =>
+          p.writeString(formatter.format(v))
+        case (StandardType.ZonedDateTime(formatter), v: ZonedDateTime) =>
+          p.writeString(formatter.format(v))
+        case (_, _) => ()
+      }
+      
+    def encodePrimitive[A](fieldNumber: Option[Short], standardType: StandardType[A], value: A): Unit = {
+      fieldNumber.foreach(num =>
+        p.writeFieldBegin(
+          new TField("", getPrimitiveType(standardType), num)
+        )
+      )
+      writePrimitiveType(standardType, value, p)
+      fieldNumber.foreach(_ =>
+        p.writeFieldEnd()
+      )
+    }
+
+    def encodeSequence[A](fieldNumber: Option[Short], schema: Schema[A], v: Chunk[A]): Unit = {
+      fieldNumber.foreach(
+        num => p.writeFieldBegin(new TField("", TType.LIST, num))
+      )
+      p.writeListBegin(new org.apache.thrift.protocol.TList(getType(schema), v.length))
+      v.foreach(encodeValue(None, schema, _))
+      p.writeListEnd();
+      fieldNumber.foreach(
+        _ => p.writeFieldEnd()
+      )
+
+    }
+
+
+    def encodeOptional[A](fieldNumber: Option[Short], codec: Schema[A], v: Option[A]): Unit =
```

Comment:
With respect to both `Option` and `List` see the recent change to the Protobuf codec (https://github.com/zio/zio-schema/pull/157). We need to be able to encode `Option[Option[_]]` and `List[List[_]]` type structures so we may need to explicitly encode `None` so we can differentiate between `Some(None)` and `None` values. Likewise for `List(Nil)` and `Nil`. Thrift support lists of lists natively unlike protobuf so this may not be a problem but something to be aware of.

Reply:
This is a problem. I do not know how to do it. When I generate java classes via `thrift` command, they do it that way, that they omit null values, do not even encode the field name for it. I do not see how I can encode `Option[Option[Int]]` and differentiate between `None` and `Some(None)`. How to do it semantically? Thrift does not have a notion of `Option[_]`, only the `null` value. Nested lists, even empty ones are working as list always has the header with its size so when I have List[List[Int]] I can differentiate between `Nil` and `List[Nil]`. There is even test for it.

Reply:
I explicitly encoded Options as in Protobuf. Plus decoder can decodes also omitted field as None (Thrift generated classes behavior).